src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -496,7 +496,10 @@ event::ptr primitive_inst::realloc_if_needed() {
 
                 _outputs[0] = variable.get_memory();
                 // To record shape predictor
-                auto prealloc_info = sp.predict_preallocation_shape(id(), _impl_params->output_layouts[0].get_shape(), dt_size, true);
+                auto prealloc_info = sp.predict_preallocation_shape(id(), cldnn::layout(_impl_params->output_layouts[0].get_shape(),
+                                                                                        actual_layout.data_type,
+                                                                                        _impl_params->output_layouts[0].format),
+                                                                    true);
```

Comment:
Can't we just use `_impl_params->output_layouts[0]` here?
```suggestion
                auto prealloc_info = sp.predict_preallocation_shape(id(), _impl_params->output_layouts[0], true);
```

Reply:
Updated as you commented, thanks.